Feedme cleanup + settings panel

Sicprofundus opened the pull request:
- bracket cleanup
- spacing cleanup
- some deprecation cleanup
- !_stricmp to ci_equals
- settings panel
- - including "click to add"

Sicprofundus committed:
FeedMe Updates

- bracket cleanup
- spacing cleanup
- some deprecation cleanup
- !_stricmp to ci_equals
- settings panel
- - including "click to add"

Sicprofundus committed:
FeedMe

Sicprofundus committed:
.

Sicprofundus committed:
Ignore Safe Zones for auto consumption

- Add ability to ignore safe zones for auto consumption

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
+bool          bFoodWarn = false;			// Announce No Food
+bool          bDrinkWarn = false;			// Announce No Drink
+bool		  bIAmCamping = false;			// Defining if we are camping out or not
+bool		  bIgnoreSafeZones = false;		// don't consume in "safe zones"
```
All the places where you converted the inline spacing to tabs, you want to switch back to spaces. The reason for that is because tab settings can be different on multiple computers, but spacing won't be.  Tabs vs spaces for alignment doesn't matter when you get to alignment after the initial indent. Always spaces (and here you now have a mix of spaces for alignment in the beginning and tabs for alignment in the end).

You can see some of this in the difference between (for example) your IAmCamping where you've used a tab and bAnnLevels where there's spaces, but I was specifically talking about between the end of the code and the comment.

Sicprofundus replied:
thanks - this just showed me that the control+r, control+w  "view white spaces" thing which shows my tabs and spaces got turned off and i didn't notice

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -79,38 +253,50 @@ void ReadList(std::list<std::string>* MyList, PCHAR fSec)
 {
-	auto pChar = GetCharInfo();
 	auto pChar2 = GetPcProfile();
+	int iZoneID = pLocalPlayer->GetZoneID() & 0x7FFF;
```
I would guard pLocalPlayer here since this GoodToConsume() function can be called from anywhere.

Also, what's the significance of 0x7FFF?

Sicprofundus replied:
the & 0x7FFF is commented in eqclasses.h as "Dynamic zones share the lower 0x7fff bits with the base zone." so it removes that bit and leaves a 15 bits for the zone id.

I'm *sure* you know this, but you're asking me if i know because i should probably use a function or offer some other clarity on what is happening. sound right?

brainiac replied:
`EQWorldData::GetZoneBaseId`

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
 
 const char* PLUGIN_NAME = "MQ2FeedMe";
 
+using namespace mq::datatypes;
```
Personally not a fan of using this shim on new code, especially if it's only to save a few characters on a couple lines. Not a huge thing but no sense using it with so few data types.

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
 
+using namespace mq::datatypes;
+
+class MQ2FeedMeType : public MQ2Type {
```
I'm guessing this wasn't tested, I'd recommend testing these datatypes (or you might not have committed the code where you did test them).

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
+
+	//~MQ2FeedMeType()
+	//{
+	//}
```
No need for this.

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
+	//{
+	//}
+
+	//virtual bool GetMember(MQVarPtr VarPtr, const char* Member, char* Index, MQTypeVar& Dest) override
```
Eh?

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
+
+class MQ2FeedMeType : public MQ2Type {
+public:
+	enum FeedMeMembers {
```
New code, you probably want enum class here.

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
+	};
+
+	MQ2FeedMeType() : MQ2Type("FeedMe") {
+		TypeMember(FeedAt);
```
Once you have enum class, these can be scoped type members.

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
+
+	//virtual bool GetMember(MQVarPtr VarPtr, const char* Member, char* Index, MQTypeVar& Dest) override
+	bool MQ2FeedMeType::GetMember(MQVarPtr VarPtr, const char* Member, char* Index, MQTypeVar& Dest) {
+		MQTypeMember* pMember = MQ2FeedMeType::FindMember(Member);
```
Genuine question because I'm rusty: Do you need MQ2FeedMeType:: here?

brainiac replied:
no

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
+			return false;
+		}
+
+		switch ((FeedMeMembers)pMember->ID) {
```
The way you've done your inline { here, but above you're on the next line and below you're in a weird place for indentation is nitpicky, but you wouldn't ask me to review if you didn't want nitpicky.  Probably want to fix your spacing.

Also, typically case is on the same indent level as switch.

And finally, what kind of cast is this and why?

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
+				return true;
+			case IgnoreSafeZones:
+				Dest.DWord = bIgnoreSafeZones;
+				Dest.Type = pBoolType;
```
Given your default return is pBoolType, you can probably just update the Type for the two ints and let pBoolType be your default here.

Also, again, rusty here, but isn't the modern nomenclature Dest.Set versus a direct set?

brainiac replied:
> Also, again, rusty here, but isn't the modern nomenclature Dest.Set versus a direct set?

yes.

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
+	bool MQ2FeedMeType::FromString(MQVarPtr& VarPtr, const char* Source)
+	{
+		return false;
+	}
```
The three of these don't appear to be doing anything.  Remove the verbosity if it's not performing an action.

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
+	//}
+
+	//virtual bool GetMember(MQVarPtr VarPtr, const char* Member, char* Index, MQTypeVar& Dest) override
+	bool MQ2FeedMeType::GetMember(MQVarPtr VarPtr, const char* Member, char* Index, MQTypeVar& Dest) {
```
Is this not an override?

brainiac replied:
should not have `MQ2FeedMeType::` in class body either. this is malformed.

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
+
+MQ2FeedMeType* pFeedMeType = nullptr;
+
+//bool TrophyData(const char* szIndex, MQTypeVar& Dest)
```
?

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
+//bool TrophyData(const char* szIndex, MQTypeVar& Dest)
+bool dataFeedMe(const char* szIndex, MQTypeVar& Dest)
+{
+	Dest.DWord = 1;
```
1?

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
+	Zone_plhdkeinteriors3a2 = 749,
+	Zone_plhdkeinteriors3a3 = 750,
+	Zone_guildhall3 = 751,
+};
```
I feel like there's gotta be a better way to do this part. Either way I probably wouldn't use an enum and if you were going to use an enum then it should probably be scoped.

But this is something I'd think there's a method of determining at runtime and if not, is this plugin really the place to store this information?

Sicprofundus replied:
I thought about that a good bit - this is one of those things where i think people will "disagree" as what constitutes a "safe zone" for purposes of consumption 'out in the open', so i opted to go this route

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -342,23 +684,131 @@ PLUGIN_API void SetGameState(DWORD GameState)
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Announce Levels and Consumption.");
+
+	if (ImGui::Checkbox("IgnoreSafeZones", &bIgnoreSafeZones))
```
You probably want spaces here given this is at the UI level.  But I would also change it to something more human readable like you did with your other settings.

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
+		default:
+			return false;
+	}
+}
```
And this function is the reason that even a scoped enum probably doesn't make sense. You could use some other form of set/list and not have to add it in both places every time...and just have a function that checks if it's in the set.  But, again, probably not the place for this.

brainiac replied:
What problem is `IsSafeZone` trying to solve?

Sicprofundus replied:
> What problem is `IsSafeZone` trying to solve?

this is a highly complained about behavior - people sitting in guild hall, lobby, poknowledge, etc just "chomp chomp chomp" while clearly afk - absolutely clearly automated.

in a super perfect world, I'd like to move our topmost food/drink to somewhere else in your bags, so you wouldn't auto consume food on top of not plugin consuming food/drink while in the safe zones - but with all the various scripts and things people run manipulating things on cursor, i decided just turning it off at least helps mask the behavior for folks that want to do that.

the only reason i didn't submit it as defaulted to true, is that it is behavior changing, and it has the potential of having someone be upset that their expensive food was consumed instead of eating their fish/lard/whatever

brainiac replied:
Try using `pZoneInfo->bNoBuffExpiration`

brainiac replied:
Or maybe `pLocalPlayer->bBuffTimersOnHold` is better.

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -79,38 +253,50 @@ void ReadList(std::list<std::string>* MyList, PCHAR fSec)
 	MyList->clear();
-	if(GetPrivateProfileString(fSec,NULL,"",Buffer,MAX_STRING*10,INIFileName)) {
+
+	if(GetPrivateProfileString(fSec, NULL, "", Buffer, MAX_STRING * 10 , INIFileName))
```
You break it, you bought it. Why are we doing this if we're cleaning it up?  Why are we using MAX_STRING * 10?

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -79,38 +253,50 @@ void ReadList(std::list<std::string>* MyList, PCHAR fSec)
+				MyList->push_back(std::string(szTemp));
+			}
+
+			pBuffer += strlen(pBuffer) + 1;
```
This section seems ugly and archaic. Pretty sure we have much better methods of doing this now.

brainiac replied:
We do. `GetPrivateProfileKeys` would replace all of this I believe.

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -79,38 +253,50 @@ void ReadList(std::list<std::string>* MyList, PCHAR fSec)
+			GetPrivateProfileString(fSec, pBuffer, "", szTemp, MAX_STRING, INIFileName);
+			if (szTemp[0] != 0)
+			{
+				MyList->push_back(std::string(szTemp));
```
When should you use push_back and when should you use emplace_back?

brainiac replied:
this should be emplace_back.

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -79,38 +253,50 @@ void ReadList(std::list<std::string>* MyList, PCHAR fSec)
+		&& !WindowOpen("BigBankWnd") && !WindowOpen("BankWnd")	// not banking
+		&& !WindowOpen("LootWnd")								// not looting
+		&& pLocalPC->pSpawn->StandState != STANDSTATE_FEIGN		// not Feigned
+		&& (bIgnoreSafeZones ? !IsSafeZone(iZoneID) : true)		// if we are ignoring safe zones, make sure we're not in one
```
This seems a little weird to me here. I'd probably write it as:
```
&& (!bIgnoreSafeZones || !IsSafeZone(iZoneID))
```

Rather than using a ternary and the "true"

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
 	char zOutput[MAX_STRING]={0}; va_list vaList; va_start(vaList,zFormat);
-	vsprintf_s(zOutput,zFormat,vaList); if(!zOutput[0]) return;
-	DoCommand(GetCharInfo()->pSpawn,zOutput);
+	vsprintf_s(zOutput, zFormat, vaList);
```
You cleaned this line up but not the line above. But I would question whether you need an Execute function at all (and based on your change below I believe you removed the only usage of it already).

brainiac replied:
This whole function can be deleted and calls can be replaced with `DoCommandf`

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
+	{
+		if (!pItem->GetItemDefinition()->FoodDuration)
+		{
+			WriteChatf("%s:: \arThat's not food or drink. Don't be rediculous", PLUGIN_NAME);
```
Is this when Redbot is being Ridiculous?

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
+{
+	if (!ItemOnCursor())
+	{
+		WriteChatf("%s:: \arNeed to have a food item on your cursor to do this.", PLUGIN_NAME);
```
Food item? In the drink section? Localized entirely in this function? At this time of year?

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
+	GenericCommand(szLine);
+}
+
+void AddDrinkItem()
```
Why do we handle food but add drink? (I'll be honest, I didn't look too closely at either one nor did I compare the old one to the new)

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
-		if (bAnnLevels) WriteChatf("\ay%s\aw:: Current Thirst(\ag%d\ax) Hunger(\ag%d\ax)", PLUGIN_NAME, GetPcProfile()->thirstlevel, GetPcProfile()->hungerlevel);
+}
+
+void AutoDrinkCmd(PlayerClient* pLPlayer, const char* szLine)
```
Does it seem like the methodology for AutoDrink and AutoFeed are going to be quite identical save for the item and checks that you're doing? If it is, you might want to go one more layer of abstraction here rather than having both do the same thing with similar verbosity.

Of course, if the plan is that they diverge that's not necessarily the case.

Sicprofundus replied:
frankly I'd prefer a singular `/feedme` command and not have /autofeed and /autodrink but i don't want to change behavior so much that it screws things up for folks

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
+			iDrinkAt = 5000;
+		}
+		sprintf_s(Buffer, "%d", iDrinkAt);
+		WritePrivateProfileString(GetCharInfo()->Name, "AutoDrink", Buffer, INIFileName);
```
Why is Buffer used here?

brainiac replied:
why is GetCharInfo() used here?

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
+		ListTypes(Thirst);
+		if (bAnnLevels)
+		{
+			sprintf_s(Buffer, "%d", iDrinkAt);
```
Why is a Buffer used here?

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
 {
-	if (!IAmCamping && strstr(Line, "It will take you about 30 seconds to prepare your camp.")) {
-		IAmCamping = true;
+	if (!bIAmCamping && ci_find_substr(Line, "It will take you about 30 seconds to prepare your camp.") != -1)
```
Why is this case insensitive?

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
 	}
-	else if (IAmCamping && strstr(Line, "You abandon your preparations to camp.")) {
-		IAmCamping = false;
+	else if (bIAmCamping && ci_find_substr(Line, "You abandon your preparations to camp.") != -1)
```
Also...why is THIS case insensitive? (But also you have an if and an else if that are specifically checking the same variable, so you probably have something you can do here to tighten this up instead of checking it in succession).

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
+	if (bIAmCamping && GetGameState() == GAMESTATE_INGAME)
+	{
+		bIAmCamping = false;
+	}
```
Okay, but like... what if you just know you're not camping if you zone? The plugin only works in Game anyway.

brainiac replied:
`if (gGameState == GAMESTATE_CHARSELECT) bIAmCamping = false`?

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
+		if (GetPcProfile())
+		{
+			iDrinkAt     = GetPrivateProfileInt(GetCharInfo()->Name, "AutoDrink", 0, INIFileName);
+			iFeedAt     = GetPrivateProfileInt(GetCharInfo()->Name, "AutoFeed",  0, INIFileName);
```
Either take out the spaces like the other lines (inconsistently) do OR put them all the same distance for alignment. But keeping the spaces AND making them out of alignment is just being mean to your reader.

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -342,23 +684,131 @@ PLUGIN_API void SetGameState(DWORD GameState)
-			if (!Loaded) {
-				ReadList(&Hunger,"FOOD");
-				ReadList(&Thirst,"DRINK");
+			WritePrivateProfileString("Settings", "IgnoreSafeZones", bIgnoreSafeZones ? "1" : "0", INIFileName);
```
What? There's no siren emoji?

Really? There's no red flag emoji?

👃 

What WritePrivateProfile could be used here that's not String...

Sicprofundus replied:
:rotating_light:

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -342,23 +684,131 @@ PLUGIN_API void SetGameState(DWORD GameState)
+		WritePrivateProfileBool("Settings", "IgnoreSafeZones", bIgnoreSafeZones, INIFileName);
+	}
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Ignore Safe Zones like poknowledge or guild lobby for auto consumption.");
```
I'd probably not use a short name with an uncapitalized long name, just for consistency sake.

Sicprofundus replied:
nice, wouldn't have caught that without you

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -342,23 +684,131 @@ PLUGIN_API void SetGameState(DWORD GameState)
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Ignore Safe Zones like poknowledge or guild lobby for auto consumption.");
+
+	if (ImGui::Checkbox("Food Warn", &bFoodWarn))
```
Is "Food Warn" really the best UI tag for this?

Sicprofundus replied:
well, I was hoping to have your avatar as a gif/emoji with a <('You are out of Food') but a slightly better collection of words might work

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -342,23 +684,131 @@ PLUGIN_API void SetGameState(DWORD GameState)
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Warn when no food.");
+
+	if (ImGui::Checkbox("Drink Warn", &bFoodWarn))
```
Is "Drink Warn" really the best UI tag for this?

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -342,23 +684,131 @@ PLUGIN_API void SetGameState(DWORD GameState)
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Warn when no drink.");
+
+	if (ImGui::InputInt("DrinkAt", &iDrinkAt)) {
```
Is "DrinkAt" really the best UI tag for this?

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -342,23 +684,131 @@ PLUGIN_API void SetGameState(DWORD GameState)
+		WritePrivateProfileInt("Settings", "AutoDrink", iDrinkAt, INIFileName);
+	}
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Thirst level to drink.");
```
I'd recommend adding some more info here about what the different numbers mean.

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -342,23 +684,131 @@ PLUGIN_API void SetGameState(DWORD GameState)
+		WritePrivateProfileInt("Settings", "AutoFeed", iFeedAt, INIFileName);
+	}
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Hunger level to eat.");
```
I'd recommend adding some more info here about hwat the different numbers mean.

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -342,23 +684,131 @@ PLUGIN_API void SetGameState(DWORD GameState)
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Thirst level to drink.");
+
+	if (ImGui::InputInt("FeedAt", &iFeedAt)) {
```
Probably could stand to rename this UI element too.

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -342,23 +684,131 @@ PLUGIN_API void SetGameState(DWORD GameState)
+	mq::imgui::HelpMarker("Hunger level to eat.");
+
+	const float buttonsize = ImGui::GetWindowSize().x * 0.45f;
+	if (ImGui::Button("Add Food From Cursor", ImVec2(buttonsize, 0)))
```
This sounds like a setting when you want it to be an action button. So Maybe "Add Food Currently on Cursor"

Knightly1 commented on MQ2FeedMe.cpp:
```diff
@@ -342,23 +684,131 @@ PLUGIN_API void SetGameState(DWORD GameState)
+	}
+	ImGui::SameLine();
+
+	if (ImGui::Button("Add Drink On Cursor", ImVec2(buttonsize, 0)))
```
This sounds like a setting when you want it to be an action button. So Maybe "Add Drink Currently on Cursor"

Knightly1 reviewed (changes requested):
I did not review the whole thing, but this should give you some stuff to work on.

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
+			return false;
+		}
+
+		if (!pLocalPlayer)
```
nothing uses `pLocalPlayer` here, no need to check for it.

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
-		if (GetPcProfile()) {
-			DrnkAt     = GetPrivateProfileInt(GetCharInfo()->Name, "AutoDrink", 0, INIFileName);
-			FeedAt     = GetPrivateProfileInt(GetCharInfo()->Name, "AutoFeed",  0, INIFileName);
+		if (GetPcProfile())
```
GetPcProfile doesn't feel like the right thing to be checking here.

Sicprofundus replied:
I'm not sure why that was originally added there - it seems if we're already in game we should be fine.

Unless you have something specific you think is more appropriate I'm inclined to remove it

brainiac replied:
well its using GetCharInfo() below. (which should be pLocalPC, but I digress...)

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -342,23 +684,131 @@ PLUGIN_API void SetGameState(DWORD GameState)
+			ImGui::PushID(static_cast<int>(index));
+			ImGui::TableNextRow();
+			ImGui::TableNextColumn();
+			ImGui::Text(food.c_str());
```
`ImGui::Text("%s", food.c_str());` or use ImGui::TextUnformatted

Text expects a format string. If food had a `%` character in it, it could be bad.

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
+int			 iDrinkAt = 0;					// Drink Level
+char         Buffer[16] = {0};
+std::list<std::string> Hunger;				// Hunger Fix List
+std::list<std::string> Thirst;				// Thirst Fix List
```
since we're here and changing everything, this should be a std::vector

Also, Thirst and Hunger are such bad names for variables of list of things to eat and drink.

Sicprofundus replied:
yeah haven't used std::list much (if at all), i originally went to move it to a vector, but the usage of with ReadList and its application wasn't clear to me. I'll take a look

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,201 @@
-std::list<std::string> Thirst;         // Thirst Fix List
+int			 iFeedAt = 0;					// Feed Level
+int			 iDrinkAt = 0;					// Drink Level
+char         Buffer[16] = {0};
```
Does anything even use this? Should it?

Sicprofundus replied:
it is used - it shouldn't exist, no

Sicprofundus committed:
Fix space vs tab

Sicprofundus committed:
Type Member update/fix

Sicprofundus committed:
Guard with pLocalPlayer

Sicprofundus committed:
Remove Unneeded check

Sicprofundus committed:
fix/remove usage of Buffer

Sicprofundus committed:
Fix potential issue if food/drink had % in it

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
 
 void Consume(uint8_t itemClass, const std::list<std::string>& fLIST)
 {
 	for (const std::string& name : fLIST) {
```
if you're going to change the braces in this function, fix it here too

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
+			WritePrivateProfileString("Drink", fmt::format("Drink{}", DrinkIndex), pItem->GetName(), INIFileName);
+			Thirst.push_back(pItem->GetName());
+		}
+		else if (!pItem->GetItemDefinition()->FoodDuration)
```
this check makes no sense. If it doesn't have a food duration it complains. if it does it will still complain. what is its purpose?

Sicprofundus replied:
yeah we already check if it doesn't have a food duration, so this is superfluous and redundant

Sicprofundus committed:
ZoneID

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
 {
 	static int Pulses = 0;
-	if (++Pulses < 50) return;
+	if (++Pulses < 50)
```
i love framerate dependent logic!

Sicprofundus committed:
Fix

- Remove redundant (and not reachable) !food duration check

Sicprofundus committed:
Fix

- GetPcProfile check to pLocalPC
- GetCharInfo() to pLocalPC

Sicprofundus committed:
bracket

- fixed braces to match style

Sicprofundus committed:
review

- removed spaces

Sicprofundus committed:
WritePrivateProfile Int not string

Sicprofundus committed:
Betterer UI tags

- provide a slightly better descriptor for the UI checkboxes

Sicprofundus committed:
pr update

Sicprofundus committed:
clean up text / descriptions

Sicprofundus committed:
button text cleanup

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -38,27 +38,192 @@
+
+	MQ2FeedMeType() : MQ2Type("FeedMe")
+	{
+		ScopedTypeMember(FeedMeMembers, FeedAt);
```
this is intended to be used with a scoped enum (i.e. `enum class`). You should update FeedMeMembers to be `enum class` and then update the switch case labels below with `FeedMeMembers::`

Sicprofundus replied:
gotcha - thanks

Sicprofundus committed:
Fix type memebers

- fixed type members to use scoped enum

Sicprofundus committed:
FeedMe Updates

- changed list to vector
- "Hunger" to vFoodList
- "Thirst" to vDrinkList
- HandleFoodDrinkItem to HandleAddFoodDrinkItem
- add ability to double click to remove an item from the Food/Drink lists
- added which ini section for helpmarkers
- adjusted text for clarity

Sicprofundus committed:
.

Sicprofundus committed:
FeedMe

- fix deref warning

Sicprofundus committed:
pulse delay

- Added a 1 second pulse delay
- removed the frame dependent delay

Sicprofundus committed:
Updates

- Change IgnoreSafeZones by CLI
- adjust size and center the text for double clicking to remove an item from the list

Sicprofundus committed:
FeedMe review updates

- I *believe* i got all the feedback thusfar
- - some items github had minimized and i didn't see until after I had already made a bunch of changes, so if I skipped something please let me know

Sicprofundus committed:
FeedMe

- use bBuffTimersOnHold instead of a curated list of zones
- PCHAR to const char*

Sicprofundus committed:
Merge remote-tracking branch 'origin/master' into update

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -75,49 +153,72 @@ bool AbilityInUse()
+			sizeof(value),
+			INIFileName
+		);
+		vVector.push_back(value);
```
This entire function looks like a re-implementation of GetPrivateProfileKeys, but your buffer is sorely undersized.

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -75,49 +153,72 @@ bool AbilityInUse()
+	if(GetGameState() == GAMESTATE_INGAME						// currently ingame
+		&& pLocalPC												// have Charinfo
+		&& pLocalPC->pSpawn										// have a Spawn
+		&& pChar2												// have PcProfile
```
don't need to check this.

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -75,49 +153,72 @@ bool AbilityInUse()
+
+	if(GetGameState() == GAMESTATE_INGAME						// currently ingame
+		&& pLocalPC												// have Charinfo
+		&& pLocalPC->pSpawn										// have a Spawn
```
don't need to check this.

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -75,49 +153,72 @@ bool AbilityInUse()
+		&& pLocalPC->pSpawn										// have a Spawn
+		&& pChar2												// have PcProfile
+		&& !ItemOnCursor()										// nothing on cursor
+		&& (!IsCasting() || pChar2->Class == Bard)				// not casting unless bard
```
just use pLocalPC->GetClass()

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -75,49 +153,72 @@ bool AbilityInUse()
+		&& !WindowOpen("TradeWnd")								// not trading with someone
+		&& !WindowOpen("BigBankWnd") && !WindowOpen("BankWnd")	// not banking
+		&& !WindowOpen("LootWnd")								// not looting
+		&& pLocalPC->pSpawn->StandState != STANDSTATE_FEIGN		// not Feigned
```
pLocalPlayer is the same as pLocalPC->pSpawn. Prefer using pLocalPlayer.

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+		return;
+	}
+
+	DoCommand(zOutput);
```
delete Execute and just use DoCommandf.

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
 }
 
-void Consume(uint8_t itemClass, const std::list<std::string>& fLIST)
+void Consume(uint8_t itemClass, std::vector<std::string> &vVector)
```
pass vVector by const ref. rename it to something more useful.

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+		auto pItemClass = pItem->GetItemClass();
+		if (pItemClass == ItemClass_Food)
+		{
+			int FoodIndex = static_cast<int>(vFoodList.size()) + 1;
```
we don't really need to calculate this ahead of time. insert the item then get the index.

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+		return;
+	}
+
+	std::vector<std::string>* targetVector = nullptr;
+	std::string sectionName;
+
+	if (ci_equals(type, "drink"))
+	{
+		targetVector = &vDrinkList;
+		sectionName = "Drink";
+	}
+	else if (ci_equals(type, "food"))
+	{
+		targetVector = &vFoodList;
+		sectionName = "Food";
+	}
```
this is a good approach. we could reuse it for the Add function above.

brainiac replied:
n/m that doesn't make sense to do, add works based on item type not the param. ignore that idea!

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+		for (int i = 0; i < static_cast<int>(targetVector->size()); ++i)
+		{
+			WritePrivateProfileString(sectionName.c_str(),
+				fmt::format("{}{}",
+					sectionName,
+					i).c_str(),
```
```suggestion
				fmt::format("{}{}", sectionName, i).c_str(),
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+			{
+				bIgnoreSafeZones = false;
+			}
+			else {
```
```suggestion
			else
			{
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
 	GetArg(Arg, szLine, 1);
-	if (!_stricmp(Arg, "list")) {
+
+	if (ci_equals(Arg, "list")) {
```
```suggestion
	if (ci_equals(Arg, "list"))
	{
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+	}
+	else if (IsNumber(Arg))
+	{
+		iFeedAt = GetIntFromString(Arg, 5000);
+
+		if (iFeedAt < 0)
+		{
+			iFeedAt = 0;
+		}
+		else if (iFeedAt > 5000)
+		{
+			iFeedAt = 5000;
+		}
```
```suggestion
		iFeedAt = std::clamp(GetIntFromString(Arg, 5000), 0, 5000);
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+			iFeedAt = 5000;
+		}
+
+		WritePrivateProfileInt(GetCharInfo()->Name, "AutoFeed", iFeedAt, INIFileName);
```
```suggestion
		WritePrivateProfileInt(pLocalPC->Name, "AutoFeed", iFeedAt, INIFileName);
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -75,49 +153,72 @@ bool AbilityInUse()
 {
 	int i = 1;
-	for (const std::string& value : list)
+	for (const std::string& value : vVector)
```
this is probably one of the cases where it actually makes sense to iterate with index. we're basically keeping track of one, might as well use it.

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+		// Delete any additional keys that exist
+		// this does an arbitrary check 10 past the vector size
+		// adding a function to count ini section keys seems a bit excessive
+		for (int i = static_cast<int>(targetVector->size()); i < static_cast<int>(targetVector->size()) + 10; ++i)
```
```suggestion
		for (size_t i = targetVector->size(); i < targetVector->size() + 10; ++i)
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+		for (int i = static_cast<int>(targetVector->size()); i < static_cast<int>(targetVector->size()) + 10; ++i)
+		{
+			std::string keyName = fmt::format("{}{}", sectionName, i);
+			WritePrivateProfileString(sectionName.c_str(), keyName.c_str(), NULL, INIFileName);
```
```suggestion
			DeletePrivateProfileKey(sectionName, keyName, INIFileName);
```

Sicprofundus replied:
neat, i didn't know about this one

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+		}
+		WriteChatf("\ay%s\aw:: IgnoreSafeZones (\ag%s\ax).", PLUGIN_NAME, bIgnoreSafeZones ? "\agOn" : "\arOff");
+	}
+	else if (!_stricmp("ui", Arg) || !_stricmp("gui", Arg))
```
```suggestion
	else if (ci_equals("ui", Arg) || ci_equals("gui", Arg))
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
 {
-	if (!strlen(szLine)) {
-		if (GoodToFeed()) Consume(ItemClass_Drink, Thirst);
+	if (!strlen(szLine))
```
```suggestion
	if (szLine[0] != 0)
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+	{
+		if (bFoodWarn)
+		{
+			bFoodWarn = 0;
```
```suggestion
			bFoodWarn = false;
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+		}
+		else
+		{
+			bFoodWarn = 1;
```
```suggestion
			bFoodWarn = true;
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+	}
+	else if (ci_equals(Arg, "remove"))
+	{
+		char item[MAX_STRING] = { 0 };
+		strcpy_s(item, GetNextArg(szLine));
+		if (item[0] == '\0')
```
```suggestion
		const char* item = GetNextArg(szLine);
		if (item[0] == '\0')
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
-			WriteChatf("%s:: \arNeed to have a food item on your cursor to do this.", PLUGIN_NAME);
+	else if (ci_equals(Arg, "remove"))
+	{
+		char item[MAX_STRING] = { 0 };
+		strcpy_s(item, GetNextArg(szLine));
+		if (item[0] == '\0')
```
```suggestion
		const char* item = GetNextArg(szLine);
		if (item[0] == '\0')
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+		else if (iDrinkAt > 5000)
+		{
+			iDrinkAt = 5000;
+		}
```
I can't suggest across these lines because of the way this split the diff, but use std::clamp here too.

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+			iDrinkAt = 5000;
+		}
+
+		WritePrivateProfileInt(GetCharInfo()->Name, "AutoDrink", iDrinkAt, INIFileName);
```
```suggestion
		WritePrivateProfileInt(pLocalPC->Name, "AutoDrink", iDrinkAt, INIFileName);
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+		}
+
+		WritePrivateProfileInt(GetCharInfo()->Name, "AutoDrink", iDrinkAt, INIFileName);
+		WriteChatf("\ay%s\aw:: AutoDrink (\ag%s\ax).", PLUGIN_NAME, iDrinkAt ? std::to_string(iDrinkAt).c_str() : "\aroff");
```
The way this is overloaded is unintuitive and not great. Only do it with thte parameter types are the same on both sides of the condition and you don't have to jump through hoops to make it work.

```suggestion
		if (iDrinkAt)
		{
			WriteChatf("\ay%s\aw:: AutoDrink (\ag%d\ax).", PLUGIN_NAME, iDrinkAt);
		}
		else
		{
			WriteChatf("\ay%s\aw:: AutoDrink (\aroff\ax).", PLUGIN_NAME);
		}
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
-	}
-	else if (IAmCamping && strstr(Line, "You abandon your preparations to camp.")) {
-		IAmCamping = false;
+	if (Color == USERCOLOR_DEFAULT) {
```
```suggestion
	if (Color == USERCOLOR_DEFAULT)
	{
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
-PLUGIN_API void SetGameState(DWORD GameState)
+PLUGIN_API void SetGameState(const int GameState)
 {
 	if (GameState == GAMESTATE_INGAME) {
```
```suggestion
	if (GameState == GAMESTATE_INGAME)
	{
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
-				ReadList(&Thirst,"DRINK");
+		if (pLocalPC)
+		{
+			iDrinkAt     = GetPrivateProfileInt(pLocalPC->Name, "AutoDrink", 0, INIFileName);
+			iFeedAt     = GetPrivateProfileInt(pLocalPC->Name, "AutoFeed",  0, INIFileName);
+			bAnnLevels = GetPrivateProfileInt("Settings", "Announce",  1, INIFileName) != 0;
+			bAnnConsume = GetPrivateProfileInt("Settings", "Announce", 1, INIFileName) != 0;
+			bFoodWarn  = GetPrivateProfileInt("Settings", "FoodWarn",  0, INIFileName) != 0;
+			bDrinkWarn = GetPrivateProfileInt("Settings", "DrinkWarn",  0, INIFileName) != 0;
```
```suggestion
			iDrinkAt = GetPrivateProfileInt(pLocalPC->Name, "AutoDrink", 0, INIFileName);
			iFeedAt = GetPrivateProfileInt(pLocalPC->Name, "AutoFeed",  0, INIFileName);
			bAnnLevels = GetPrivateProfileInt("Settings", "Announce",  1, INIFileName) != 0;
			bAnnConsume = GetPrivateProfileInt("Settings", "Announce", 1, INIFileName) != 0;
			bFoodWarn = GetPrivateProfileInt("Settings", "FoodWarn",  0, INIFileName) != 0;
			bDrinkWarn = GetPrivateProfileInt("Settings", "DrinkWarn",  0, INIFileName) != 0;
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+		WritePrivateProfileBool("Settings", "Announce", bAnnConsume, INIFileName);
+	}
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Announce Levels and Consumption.\n\nINI Setting: Announce");
```
This help marker text should literally be the label.

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Warn when no food.\n\nINI Setting: FoodWarn");
+
+	if (ImGui::Checkbox("Warning: No Drink", &bFoodWarn))
```
"Warn when out of drink"

Sicprofundus replied:
also lazy copy pasta ! I'm so ashamed :'(

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Ignore Safe Zones like \"poknowledge\" or \"guildlobby\" for auto consumption.\nThis does *NOT* move food around in your bags.\n\nINI Setting: IgnoreSafeZones");
+
+	if (ImGui::Checkbox("Warning: No Food", &bFoodWarn))
```
"When when out of food"

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+		ImGui::TextUnformatted("Consume Drink at thirst level:");
+		ImGui::TableNextColumn();
+		ImGui::SetNextItemWidth(iInputWidth);
+		if (ImGui::InputInt("##iDrinkAt", &iDrinkAt)) {
```
```suggestion
		if (ImGui::InputInt("##iDrinkAt", &iDrinkAt))
		{
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+		ImGui::TextUnformatted("Consume Food at hunter level:");
+		ImGui::TableNextColumn();
+		ImGui::SetNextItemWidth(iInputWidth);
+		if (ImGui::InputInt("##iFeedAt", &iFeedAt)) {
```
```suggestion
		if (ImGui::InputInt("##iFeedAt", &iFeedAt))
		{
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+	}
+
+	const float buttonsize = ImGui::GetWindowSize().x * 0.45f;
+	if (ImGui::Button("Add Food Currently on Cursor", ImVec2(buttonsize, 0)))
```
Might be able to shorten this a little bit with "Add Food From Cursor"

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+
+		ImGui::EndTable();
+	}
+	ImGui::EndChild();
```
```suggestion
	if (ImGui::BeginTable("##FeedMeFoodList", 1, ImGuiTableFlags_BordersInnerV | ImGuiTableFlags_ScrollY, ImVec2(0, 150)))
	{
		ImGui::TableSetupColumn("Food");
		ImGui::TableSetupScrollFreeze(0, 1);
		ImGui::TableHeadersRow();
		for (const std::string& item : vFoodList)
		{
			ImGui::PushID(item.c_str());
			ImGui::TableNextRow();
			ImGui::TableNextColumn();
			ImGui::TextUnformatted(item.c_str());

			// double click to remove
			if (ImGui::IsItemHovered() && ImGui::IsMouseDoubleClicked(0))
			{
				HandleRemoveFoodDrinkItem("food", item.c_str());
			}

			ImGui::PopID();
		}

		ImGui::EndTable();
	}
```

brainiac replied:
Oh that didn't diff very well. Remove the child, use ImGuiTableFlags_ScrollY instead. Remove the int casts, etc.

Do the same below.

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+		ImGui::TableSetupColumn("Drink");
+		ImGui::TableSetupScrollFreeze(0, 1);
+		ImGui::TableHeadersRow();
+		for (int i = 0; i < static_cast<int>(vDrinkList.size()); i++)
```
```suggestion
		for (size_t i = 0; i < vDrinkList.size(); i++)
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -38,28 +38,106 @@
 //****************************************************************//
 
 #include <mq/Plugin.h>
+#include <mq/imgui/ImGuiUtils.h>
```
```suggestion
#include "mq/Plugin.h"
#include "mq/imgui/ImGuiUtils.h"
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+			bFoodWarn  = GetPrivateProfileInt("Settings", "FoodWarn",  0, INIFileName) != 0;
+			bDrinkWarn = GetPrivateProfileInt("Settings", "DrinkWarn",  0, INIFileName) != 0;
+
+			WritePrivateProfileInt("Settings", "Announce", bAnnLevels ? 1 : 0, INIFileName);
+			WritePrivateProfileInt("Settings", "FoodWarn", bFoodWarn ? 1 : 0, INIFileName);
+			WritePrivateProfileInt("Settings", "DrinkWarn", bDrinkWarn ? 1 : 0, INIFileName);
+			WritePrivateProfileInt("Settings", "IgnoreSafeZones", bIgnoreSafeZones ? 1 : 0, INIFileName);
```
```suggestion
			WritePrivateProfileBool("Settings", "Announce", bAnnLevels, INIFileName);
			WritePrivateProfileBool("Settings", "FoodWarn", bFoodWarn, INIFileName);
			WritePrivateProfileBool("Settings", "DrinkWarn", bDrinkWarn, INIFileName);
			WritePrivateProfileBool("Settings", "IgnoreSafeZones", bIgnoreSafeZones, INIFileName);
```

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -38,28 +38,106 @@
-
-bool IAmCamping = false;               // Defining if we are camping out or not
+bool          bAnnLevels = true;            // Announce Levels
+bool          bAnnConsume = true;           // Announce Consumption
```
Who is Ann and why do we want to consume her?

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+	if (!type)
+	{
+		return;
+	}
```
Remove. Unnecessarily implies `type` could be `nullptr`, but it never is.

brainiac commented on MQ2FeedMe.cpp:
```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+	// update ini with our vector of drinks
+	if (targetVector)
+	{
+		for (int i = 0; i < static_cast<int>(targetVector->size()); ++i)
```
```suggestion
		for (size_t i = 0; i < targetVector->size(); ++i)
```

Sicprofundus committed:
Update includes

- changed includes from <foo> to "foo"
- - learned that <foo> should be for standard libraries or third part libraries, and "foo" should be for user/project

Sicprofundus committed:
Updates from review comments

- AnnLevels to AnnounceLevels
- AnnConsume to AnnounceConsume
- use PLUGINMSG
- adjust text output for readability
- !_stricmp to ci_equals
- utilize std::clamp

Sicprofundus committed:
Brackets + Spacing

- Updated brackets to match styling
- updated tabs/spacing

Sicprofundus committed:
Update Loops

- updated ListTypes loop per review feedback
- update targetVector loops
- utilized DeletePrivateProfileKey

Sicprofundus committed:
Remove vestigial function

- removed the ever ominous sounding "Execute" function that was no longer in use

Sicprofundus committed:
UI Cleanup

- adjusted capitalization usage
- clarified checkbox titles
- fixed brackets to match styling
- adjusted the food and drink list tables to not use a child and instead a ScrollY

Sicprofundus committed:
Spacing

- Fixed spacing

Sicprofundus committed:
Typo

- corrected "hunter" to "hunger
